### PR TITLE
Fix live and upcoming community videos

### PR DESCRIPTION
**Pull request by ChunkyProgrammer**
closes https://github.com/iv-org/invidious/issues/3731

This pr does the following:

- skips video length + publish date if they don't exist
- parses owner verified
- parse liveNow and isUpcoming
- parses viewcount runs if simple text is not available

### CODE REVIEW

File: src/invidious/channels/community.cr [hunk 1]
```diff
@@ -155,14 +157,22 @@ def fetch_channel_community(ucid, continuation, locale, format, thin_mode)
...
                         # TODO: json.field "authorThumbnails", "channelThumbnailSupportedRenderers"
-                        # TODO: json.field "authorVerified", "ownerBadges"
+                        json.field "authorVerified", has_verified_badge?(attachment["ownerBadges"]?).to_s
 
-                        published = decode_date(attachment["publishedTimeText"]["simpleText"].as_s)
+                        if published_time_text = attachment.dig?("publishedTimeText", "simpleText").try &.as_s
+                          published = decode_date(published_time_text)
+                          json.field "published", published.to_unix
+                          json.field "publishedText", translate(locale, "`x` ago", recode_date(published, locale))
+                        end
```
Comment (SamantazFox) [inline src/invidious/channels/community.cr:166]: For compatibility, can you please keep the fields, but put dummy values in here (0 for timestamp, and empty string for text). I'd prefer if we don't hard break other downstream projects!

File: src/invidious/channels/community.cr [hunk 2]
```diff
@@ -155,14 +157,22 @@ def fetch_channel_community(ucid, continuation, locale, format, thin_mode)
...
 
-                        json.field "published", published.to_unix
-                        json.field "publishedText", translate(locale, "`x` ago", recode_date(published, locale))
+                        if thumbnail_overlay = attachment.dig?("thumbnailOverlays", 0, "thumbnailOverlayTimeStatusRenderer")
+                          overlay_style = thumbnail_overlay["style"].as_s
+                          json.field "liveNow", overlay_style == "LIVE"
+                          json.field "isUpcoming", overlay_style == "UPCOMING"
+                        end
```
Comment (SamantazFox) [inline src/invidious/channels/community.cr:172]: Ditto

**Review by SamantazFox — CHANGES_REQUESTED** [review]

File: src/invidious/channels/community.cr (hunk 1 above)
Comment (ChunkyProgrammer) [reply src/invidious/channels/community.cr]: Sounds good, i made the change for the lengthSeconds as well 🙂

**Comment by ChunkyProgrammer** [comment]
Found a better way of doing this 😄

**Pull request closed without merging**